src/scheduler/workers/main.ts

```diff
@@ -148,44 +153,43 @@ export async function processList(list: List, listItemsForList: ListItemWithHist
   const today = new Date(todayDateString);
   logger.info(`Checking annual review milestone dates against today date ${today.toISOString()} - ${JSON.stringify(annualReviewKeyDates)}`);
 
-  // email the posts and providers if today = one of the annual review milestone date
-  switch (today.toISOString()) {
-    case annualReviewKeyDates?.POST_ONE_MONTH:
-      isEmailSent = isEmailSentBefore(audit, "sendOneMonthPostEmail");
-      if (!isEmailSent) {
-        await processPostEmailsForList(list, "POST_ONE_MONTH", "sendOneMonthPostEmail");
-      }
-      break;
-    case annualReviewKeyDates?.POST_ONE_WEEK:
-      isEmailSent = isEmailSentBefore(audit, "sendOneWeekPostEmail");
-      if (!isEmailSent) {
-        await processPostEmailsForList(list, "POST_ONE_WEEK", "sendOneWeekPostEmail");
-      }
-      break;
-    case annualReviewKeyDates?.POST_ONE_DAY:
+  if (today.toISOString() === annualReviewKeyDates?.START) {
+    // email posts to notify of annual review start
+    isEmailSent = isEmailSentBefore(audit as Audit, "sendStartedPostEmail");
+    if (!isEmailSent) {
+      await processPostEmailsForList(list, "START", "sendStartedPostEmail");
+    }
+
+    // update ListItem.isAnnualReview if today = the START milestone date
+    // email providers to notify of annual review start
+    isEmailSent = isEmailSentBefore(listItemAudit as Audit, "sendStartedProviderEmail");
+    if (isEmailSent) {
+      logger.info(`Annual review started email has already been sent to providers for list ${list.id}`);
+      return;
+    }
+    const updatedListItems = await updateIsAnnualReviewForListItems(listItemsForList, list);
+    const currentDateString = formatDate(addDays(today, 42));
+    await processProviderEmailsForListItems(list, updatedListItems, "START", currentDateString);
+
+  } else if (isBefore(today, new Date(annualReviewKeyDates?.START ?? ""))) {
```

Comment:
I would stay away from else ifs and nested ifs. they get quite confusing to read otherwise.

What I think you are trying to do is something like  post_one_day < today <= start? "is today's date after post_one_day, but before start date"? 

I think the statements in reverse order would make more sense, because it would then mirror the progress of time. 

i.e `POST_ONE_MONTH` should happen first, so would happen first in the code.

would `isWithinInterval` help?

`isWithinInterval(today, { start: POST_ONE_MONTH, end: POST_ONE_WEEK })`. This way you don't have to do a x < y <= z check. 

I'm also not sure if we need to pass in `today` anymore -- changing system time for support isn't necessary because we can change the keyDates object. 


in future we could probably do something clever with sorting and maps and make it completely generic. `intervals.find(todaysInterval)`


```
if (isWithinInterval(today, { start: POST_ONE_MONTH, end: POST_ONE_WEEK })) {
 


 return // to stop execution of rest of code. alternatively return the promise processPostEmailsForList so you can handle the possible error outside of this function.
}


if (isWithinInterval(today, { start: POST_ONE_WEEK, end: POST_ONE_DAY })) {

  
}

if (isToday(POST_ONE_DAY)) {
  
}

if(isToday(START)) {


}



```

Reply:
The if-else basically allows the emails closest to annual review start to be sent out if within that date range and therefore it will not send out an email for a previous date range, e.g., we dont; want one month email to be sent out if within the one week date range.  The isWithinInterval function should work for this, will take a look.

Reply:
yis. my main problem is else ifs, and nested else ifs which in my experience leads to spaghoots 

```

if(A) {

} else if(B) {
  if(C) {
  
  } else if (D) {
  
  } else if (E) {
  
  } 
}

```